src/test/java/uk/gov/pay/directdebit/pact/PublicApiContractTest.java

```diff
@@ -69,6 +73,18 @@ public void aGatewayAccountWithExternalIdAndAMandateWithExternalIdExist(Map<Stri
                 .withPayerFixture(PayerFixture.aPayerFixture())
                 .insert(app.getTestContext().getJdbi());
     }
+    
+    @State("a gateway account with external id and a confirmed mandate exists")
+    public void aGatewayAccountWithExternalIdAndAConfirmedMandateExists(Map<String, String> params) {
+        aGatewayAccountWithExternalIdAndAMandateWithExternalIdExist(params);
+        MandateDao mandateDao = app.getApplication().getInstance(MandateDao.class);
+        Mandate mandate = mandateDao.findByExternalId(MandateExternalId.valueOf(params.get("mandate_id"))).get();
+        Mandate updatedMandate = fromMandate(mandate)
+                .withMandateBankStatementReference(MandateBankStatementReference.valueOf(params.get("bank_mandate_reference")))
+                .withPaymentProviderId(GoCardlessMandateId.valueOf(params.get("unique_identifier")))
+                .build();
+        mandateDao.updateReferenceAndPaymentProviderId(updatedMandate);
```

Comment:
Why do we need the DAO to update? Can't we just add these fields when we insert using the database fixture?

Reply:
It already adds the fields when inserting using the database fixture, but I want to get rid of that because the mandate bank ref and provider id aren't available on the creation of a mandate, so https://github.com/alphagov/pay-publicapi/blob/master/src/test/resources/pacts/publicapi-direct-debit-connector-create-mandate.json#L36 is essentially incorrect and I want to rectify that after this.

Reply:
The publicapi-direct-debit-connector-create-mandate is incorrect and mandate_reference should be null.